### PR TITLE
API Integration - Registration

### PR DESCRIPTION
**Pull request by joacomiguens**
## Summary

- Create service for user registration.

## Screenshot

api-registration

## Trello Card

You can find it [here](https://trello.com/c/JSBwiXof/7-registration-api-integration)

### CODE REVIEW

File: wolox-books/src/app/global-constants.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+export const ROOT_URL = 'https://books-training-rails.herokuapp.com/api/v1';
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/global-constants.ts]: Add this url in environment file

File: wolox-books/src/app/screens/auth/screens/sign-up/sign-up.component.ts [hunk 2]
```diff
@@ -52,8 +54,10 @@ export class SignUpComponent implements OnInit {
       password_confirmation: this.signUpForm.value.password,
       locale: 'en',
     };
-    console.log(this.user);
-    this.signUpForm.reset();
-    this.router.navigate(['../auth/login'], { relativeTo: this.activatedRoute.parent });
+    this.userService.createUser(this.user).subscribe(($response) => {
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/screens/auth/screens/sign-up/sign-up.component.ts]: The `$response` variable name it's used more for observables, in this case it's just an object, use `response`.

File: wolox-books/src/app/services/user.service.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { IFullUser } from '../models/user.interface';
+import { ROOT_URL } from '../global-constants';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class UserService {
+  constructor(private readonly http: HttpClient) {}
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts:13]: Add blank sapce

File: wolox-books/src/app/services/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  baseUrl = ROOT_URL;
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts]: Also here

File: wolox-books/src/app/services/user.service.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  createUser(user: IFullUser): Observable<IFullUser> {
+    const headers = new HttpHeaders({
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts]: This header can use in other request, put on the top of the file.

File: wolox-books/src/app/services/user.service.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+      'Content-Type': 'application/json',
+    });
+    return this.http
+      .post<IFullUser>(this.baseUrl + '/users', user, { headers })
+      .pipe(
+        map(($response) => {
```
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts]: same here for the `$response` name

Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts:21]: You can delete this map and pipe operators

**Review by ebedoyaalzate — APPROVED** [review]

File: wolox-books/src/app/services/user.service.ts (hunk 6 above)
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts]: You can delete this map and pipe operators

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: wolox-books/src/app/services/user.service.ts (hunk 5 above)
Comment (ebedoyaalzate) [inline wolox-books/src/app/services/user.service.ts]: It's necessary add this header??

File: wolox-books/src/app/services/user.service.ts (hunk 6 above)
Comment (joacomiguens) [reply wolox-books/src/app/services/user.service.ts]: Done 😄

Comment (joacomiguens) [reply wolox-books/src/app/services/user.service.ts]: Perfect ✔️

File: wolox-books/src/app/global-constants.ts (hunk 1 above)
Comment (joacomiguens) [reply wolox-books/src/app/global-constants.ts]: Done 😄

**Review by ebedoyaalzate — APPROVED** [review]

File: wolox-books/src/environments/environment.ts [hunk 7]
```diff
@@ -3,7 +3,8 @@
 // The list of file replacements can be found in `angular.json`.
 
 export const environment = {
```
Comment (mcallegari10) [inline wolox-books/src/environments/environment.ts:5]: These files should be ignored by git

File: wolox-books/src/environments/environment.prod.ts [hunk 8]
```diff
@@ -1,3 +1,4 @@
 export const environment = {
-  production: true
+  production: true,
+  apiBaseUrl: 'https://books-training-rails.herokuapp.com/api/v1',
```
Comment (mcallegari10) [inline wolox-books/src/environments/environment.prod.ts:3]: These files should be ignored by git

**Review by mcallegari10 — APPROVED** [review]
Add those files to the `.gitignore` 😄

**Pull request merged**